apps/launch/components/Layout/SwitchNetworkSelect/SwitchNetworkSelect.tsx

```diff
@@ -0,0 +1,69 @@
+import { Select } from 'antd';
+import { useRouter } from 'next/router';
+import { FC } from 'react';
+import { useSwitchChain } from 'wagmi';
+
+import { useScreen } from '@autonolas/frontend-library';
+
+import {
+  ALL_SUPPORTED_CHAINS,
+  PAGES_TO_LOAD_WITHOUT_CHAINID,
+} from 'common-util/config/supportedChains';
+
+import { useHandleRoute } from './useHandleRoute';
+
+export const SwitchNetworkSelect: FC = () => {
+  const router = useRouter();
+  const { isMobile } = useScreen();
+  const { switchChainAsync } = useSwitchChain();
+  const path = router?.pathname || '';
+
+  const chainName = (router?.query?.network || 'ethereum') as string;
+
+  // handle the routing
+  useHandleRoute();
+
+  return (
+    <div style={{ marginRight: isMobile ? 8 : 0 }}>
+      <Select
+        showSearch
+        className="show-scrollbar"
+        style={{ width: isMobile ? 140 : 200 }}
+        listHeight={800}
+        value={chainName}
+        placeholder="Select Network"
+        disabled={PAGES_TO_LOAD_WITHOUT_CHAINID.some((e) => path.includes(e))}
+        options={ALL_SUPPORTED_CHAINS.map((e) => ({
```

Comment:
can we do this mapping outside of the component so it won't be recalculated on each re-render?